sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionProcessor.java

```diff
@@ -70,32 +44,36 @@ public CheckpointManager checkpointManager() {
      * asynchronously.
      *
      * <p>
-     * This is also a good place to update checkpoints as appropriate.
+     * This is also a good place to update checkpoints using the {@code partitionContext} as appropriate.
      *
+     * @param partitionContext The partition information the event data belongs to.
      * @param eventData {@link EventData} received from this partition.
      * @return a representation of the deferred computation of this call.
      */
-    public abstract Mono<Void> processEvent(EventData eventData);
+    public abstract Mono<Void> processEvent(PartitionContext partitionContext, EventData eventData);
 
     /**
      * This method is called when an error occurs while receiving events from Event Hub. An error also marks the end of
      * event data stream.
      *
+     * @param partitionContext The partition information where the error occurred.
      * @param throwable The {@link Throwable} that caused this method to be called.
      */
-    public void processError(Throwable throwable) {
-        logger.warning("Error occurred in partition processor for partition {} ", partitionContext.partitionId());
+    public void processError(PartitionContext partitionContext, Throwable throwable) {
+        logger.warning("Error occurred in partition processor for partition {} ", partitionContext.partitionId(),
```

Comment:
Does this get rendered properly? iirc the underlying slf4j .warn method does not take a varargs and then a throwable.

Reply:
This is not slf4j issue. Adding exception as the last param in slf4j `warn` method works as expected. But our `ClientLogger` removes the exception for any log level other than `verbose`. I am not sure why we do that but I prefer to have the users of `ClientLogger` pass in the exception which can contain useful information and we may change the `ClientLogger` to not remove exceptions later.